src/neuroconv/tools/spikeinterface/spikeinterface.py

```diff
@@ -1436,3 +1459,157 @@ def write_sorting(
+    template_means = waveform_extractor.get_all_templates()
+    template_stds = waveform_extractor.get_all_templates(mode="std")
+    sorting = waveform_extractor.sorting
+    if unit_ids is not None:
+        unit_indices = sorting.ids_to_indices(unit_ids)
+        template_means = template_means[unit_indices]
+        template_stds = template_stds[unit_indices]
+
+    # metrics properties (quality, template) are temporarily added as properties
+    # to be written in the Units table
+    tmp_properties = []
+    if waveform_extractor.is_extension("quality_metrics"):
+        qm = waveform_extractor.load_extension("quality_metrics").get_data()
+        for prop in qm.columns:
+            if prop not in sorting.get_property_keys():
+                sorting.set_property(prop, qm[prop])
+                tmp_properties.append(prop)
+    if waveform_extractor.is_extension("template_metrics"):
+        tm = waveform_extractor.load_extension("template_metrics").get_data()
+        for prop in tm.columns:
+            if prop not in sorting.get_property_keys():
+                sorting.set_property(prop, tm[prop])
+                tmp_properties.append(prop)
+
+    assert write_as in [
+        "units",
+        "processing",
+    ], f"Argument write_as ({write_as}) should be one of 'units' or 'processing'!"
+    if write_as == "units":
+        assert units_name == "units", "When writing to the nwbfile.units table, the name of the table must be 'units'!"
+    write_in_processing_module = False if write_as == "units" else True
+
+    with make_or_load_nwbfile(
+        nwbfile_path=nwbfile_path, nwbfile=nwbfile, metadata=metadata, overwrite=overwrite, verbose=verbose
+    ) as nwbfile_out:
+        if write_electrical_series:
+            if waveform_extractor._recording is None:
+                assert recording is not None, (
+                    "To write electrical series, the waveform_extractor needs to have a recording attached. "
+                    "For recordingless mode, you can use the 'recording' argument."
+                )
+            write_recording(waveform_extractor._recording, nwbfile_out, metadata, **write_electrical_series_kwargs)
+        else:
+            # we need add_electrodes_from_waveforms
+            if waveform_extractor._recording is None:
+                if recording is not None:
+                    assert isinstance(recording, BaseRecording), "'recording' needs to be a BaseRecording object."
+            else:
+                recording = waveform_extractor.recording
+            if recording is not None:
+                add_electrodes_info(recording, nwbfile=nwbfile_out, metadata=metadata)
+            else:
+                print(
+                    "To add the electrode table, the waveform_extractor needs to have a recording attached. "
+                    "For recordingless mode, you can use the 'recording' argument."
+                    "The electrode table will not be added."
+                )
```

Comment:
I'd use a `warnings.warn` instead of a `print` to make sure it goes to `std.err` pipe (important for loggers or other filters/captures)

Reply:
It's also a question if this ought to be a warning or an earlier assertion... It would seem odd to add waveforms, but not even have an electrodes table at all in the file (or worse, an incorrect electrodes table that only has items from a different probe)